oci/layer/tar_generate.go

```diff
@@ -219,10 +219,14 @@ func (tg *tarGenerator) AddFile(name, path string) error {
 		//       (we'd need to use libcap to parse it).
 		value, err := tg.fsEval.Lgetxattr(path, name)
 		if err != nil {
-			// XXX: I'm not sure if we're unprivileged whether Lgetxattr can
-			//      fail with EPERM. If it can, we should ignore it (like when
-			//      we try to clear xattrs).
-			return errors.Wrapf(err, "get xattr: %s", name)
+			v := errors.Cause(err)
+			log.Debugf("failure reading xattr from list on %q: %q", name, v)
+			if v != unix.EOPNOTSUPP && v != unix.ENODATA && v != unix.EPERM {
+				// XXX: I'm not sure if we're unprivileged whether Lgetxattr can
+				//      fail with EPERM. If it can, we should ignore it (like when
```

Comment:
this comment is obsolete with your change, right? (you are ignoring it now?)

Reply:
lol - yes.  in fact I didn't have EPERM in there at first, and the comment convinced me to do it.  Then I left it anyway :)

This is IMO the most controversial part, still: if we get EPERM do we want to have an error, or ignore it?

Reply:
Sounds good - updated the commit to do that.  Thanks.